tests/abi/tools/revng-ensure-rft-equivalence/Main.cpp

```diff
@@ -243,6 +243,13 @@ int main(int Argc, char *Argv[]) {
   LeftModel->ImportedDynamicFunctions().clear();
   RightModel->ImportedDynamicFunctions().clear();
 
+  // Remove all the functions starting with `__` as well.
+  auto StartsWithDoubleUnderscore = [](const model::Function &Function) {
+    return llvm::StringRef(Function.OriginalName()).substr(0, 2) == "__";
+  };
+  llvm::erase_if(LeftModel->Functions(), StartsWithDoubleUnderscore);
+  llvm::erase_if(RightModel->Functions(), StartsWithDoubleUnderscore);
+
```

Comment:
Can't we convert all the types that are directly referenced by a `model::Function` instead?

Reply:
Why? Add a comment at least.

Reply:
> Why?

I don't remember specifically. But I feel like at the time this code was written, `model diff` was the hip new thing so we were kind of pushing for it wherever applicable.

That being said, I'd say it fits here pretty well, aside from having to deal with a couple of weird cases we don't care about like this one, it does indeed do all the heavy lifting (and produces pretty nice error message as a bonus).

Reply:
OK, but we can't leave `llvm::erase_if(LeftModel->Functions(), StartsWithDoubleUnderscore)` without an explanation on why we're dropping those.

Reply:
Yeah, my bad on the explanation.

Reply:
Ping?

Reply:
Since then, I ran into similar trouble again, so it's now even stricter:
```cpp

  // Remove all the functions that are not strictly related to the test.
  auto ShouldRemove = [](const model::Function &Function) {
    llvm::StringRef Name = Function.OriginalName();
    if (Name.take_front(5) == "test_")
      return false;
    if (Name.take_front(6) == "setup_")
      return false;
    if (Name == "main")
      return false;
    return true;
  };
  llvm::erase_if(LeftModel->Functions(), ShouldRemove);
  llvm::erase_if(RightModel->Functions(), ShouldRemove);
```

Do you feel like this version also needs an even more explicit comment? (because I feel like lambda being right there makes it explicit enough)

Reply:
Works